examples/microphone_nextjs/src/utils/recorder.ts

```diff
@@ -0,0 +1,177 @@
+import { useEffect, useState } from "react";
+
+const SAMPLE_RATE_48K = 48000;
+
+// AudioRecorder is a class that wraps the MediaRecorder and device stream items
+// It also provides methods for starting and stopping recording
+export class AudioRecorder {
+  stream: MediaStream;
+  recorder: MediaRecorder;
+  audioContext: AudioContext;
+  mediaStreamSource: MediaStreamAudioSourceNode;
+  dataHandlerCallback?: (data: Float32Array | Buffer) => void;
```

Comment:
We can have this accept a `Blob` directly, since now the SDK supports this. This also prevents the need for an async call to `arrayBuffer` on line 48.